### DISCUSSION TITLE
Guides for querying, templates and roles?

### DISCUSSION THREAD

**Discussion by MadsRC** [mudler/LocalAI#956] [General]
Is there any guides or documentation anywhere on the differences between querying the regular endpoints and the openai compatible ones?

Especially on how they get mapped to templates and how roles affect those templates?

I tried futzing around with the templates and managed to find all the template names (chat_message, chat etc etc) from the source code. But I ended up, using the "chat_message" templates, just overwritng the incoming chat messages with the content of the system prompt :P

**Comment by ypankovych** [comment]
+1 here, needs better docs
